File: WebProject/wwwroot/app/Domain/Models/ExerciceComptable.ts
module app.Domain.Models {
    export interface IExerciceComptable {
        idExercice: number;
        libelleExercice: string;
        dateOuvertureExercice: Date;
        dateClotureExercice?: Date;
        dateFinExercice: Date;
        statusExercice: string;
    }

    export class ExerciceComptable extends app.Domain.GenericModel.EntityBase implements IExerciceComptable {
        constructor(public idExercice: number, public libelleExercice: string, public dateOuvertureExercice: Date, public dateFinExercice: Date, public statusExercice: string, public dateClotureExercice?: Date) {

            super();
            this.idExercice = idExercice;
            this.libelleExercice = libelleExercice;
            this.dateOuvertureExercice = dateOuvertureExercice;
            this.dateFinExercice = dateFinExercice;
            this.statusExercice = statusExercice;
            this.dateClotureExercice = dateClotureExercice;
            
        }
    }
}

File: DataModel/Models/ExerciceComptable.cs
﻿using System;
using System.Collections.Generic;

namespace DataModel.Models
{
    public partial class ExerciceComptable
    {
        public ExerciceComptable()
        {
            OperationComptable = new HashSet<OperationComptable>();
        }

        public int IdExercice { get; set; }
        public string LibelleExercice { get; set; }
        public DateTime DateOuvertureExercice { get; set; }
        public DateTime DateFinExercice { get; set; }
        public DateTime? DateClotureExercice { get; set; }
        public string StatusExercice { get; set; }

        public virtual ICollection<OperationComptable> OperationComptable { get; set; }
    }
}
